From 885b35845b92095db48eb15a8f7e68202557346a Mon Sep 17 00:00:00 2001
From: Shammah Chancellor <shammah.chancellor@gmail.com>
Date: Sun, 25 Sep 2022 12:42:49 -0700
Subject: [PATCH 175/196] [test] Use transaction sizes for checking block
 position (#274)

The current nng_interface test uses hardcoded locations in the block
data. This causes the test to break any time the transaction sizes
change due to changes in other areas of the code which generate
transactions. (e.g. a differently sized ecdsa transaction, or pad_tx
being updated)
---
 test/functional/interface_nng.py | 15 +++++++++------
 1 file changed, 9 insertions(+), 6 deletions(-)

diff --git a/test/functional/interface_nng.py b/test/functional/interface_nng.py
index ae5771dc8..7895e0a27 100755
--- a/test/functional/interface_nng.py
+++ b/test/functional/interface_nng.py
@@ -411,24 +411,27 @@ class NngInterfaceTest(BitcoinTestFramework):
         response = await self._recv_response(rpc_sock)
         response = GetBlockResponse.GetBlockResponse.GetRootAs(response, 0)
         block = response.Block()
+        tx0_raw = get_fb_bytes(block.Txs(0).Tx(), 'Raw')
+        tx1_raw = get_fb_bytes(block.Txs(1).Tx(), 'Raw')
+        tx2_raw = get_fb_bytes(block.Txs(2).Tx(), 'Raw')
+
         assert_equal(bytes(response.Block().Header().BlockHash().Hash().Data())[::-1].hex(),
                      blockhash[::-1].hex())
         assert_equal(block.MetadataLength(), 0)
         assert_equal(block.TxsLength(), 3)
         assert_equal(block.FileNum(), 0)
         assert_equal(block.UndoPos(), 4518)
-        assert_equal(block.Txs(0).DataPos(), 31561)
+        coinbase_start_position = 31561
+        assert_equal(block.Txs(0).DataPos(), coinbase_start_position)
         assert_equal(block.Txs(0).UndoPos(), 0)
         assert_equal(block.Txs(0).UndoSize(), 0)
-        assert_equal(block.Txs(1).DataPos(), 31673)
+        assert_equal(block.Txs(1).DataPos(), coinbase_start_position + len(tx0_raw))
+
         assert_equal(block.Txs(1).UndoPos(), 4519)
         assert_equal(block.Txs(1).UndoSize(), 26)
-        assert_equal(block.Txs(2).DataPos(), 31773)
+        assert_equal(block.Txs(2).DataPos(), coinbase_start_position + len(tx0_raw) + len(tx1_raw))
         assert_equal(block.Txs(2).UndoPos(), 4519 + 26)
         assert_equal(block.Txs(2).UndoSize(), 29)
-        tx0_raw = get_fb_bytes(block.Txs(0).Tx(), 'Raw')
-        tx1_raw = get_fb_bytes(block.Txs(1).Tx(), 'Raw')
-        tx2_raw = get_fb_bytes(block.Txs(2).Tx(), 'Raw')
         assert_equal(tx1_raw.hex(), tx.serialize().hex())
         assert_equal(bytes(block.Txs(1).Tx().Txid().Hash().Data())[::-1].hex(), tx.txid_hex)
         assert_equal(tx2_raw.hex(), other_tx.serialize().hex())
-- 
2.39.5 (Apple Git-154)

